broker-core/src/main/java/org/apache/qpid/server/model/adapter/FileBasedGroupProviderImpl.java

```diff
@@ -359,16 +364,21 @@ protected void onOpen()
             if (childClass == GroupMember.class)
             {
                 String memberName = (String) attributes.get(GroupMember.NAME);
-
-                _groupDatabase.addUserToGroup(memberName, getName());
-                UUID id = UUID.randomUUID();
-                Map<String,Object> attrMap = new HashMap<String, Object>();
-                attrMap.put(GroupMember.ID,id);
-                attrMap.put(GroupMember.NAME, memberName);
-                GroupMemberAdapter groupMemberAdapter = new GroupMemberAdapter(attrMap);
-                groupMemberAdapter.create();
-                return Futures.immediateFuture((C) groupMemberAdapter);
-
+                Set<String> users = _groupDatabase.getUsersInGroup(getName());
+                if(!users.contains(memberName))
```

Comment:
If a group provider is configured with caseSensitive==false, the user lookup should be case insensitive. It seems we can perform the user existence check directly in `Database#addUserToGroup(String, String)`. Thus, let's move the check there, andm throw `IllegalConfigurationException` from implementation of `Database#addUserToGroup(String, String)`.

Reply:
Done!